momepy/dimension.py

```diff
@@ -637,8 +637,20 @@ def __init__(
                 np.nanmean(rights) if ~np.isnan(rights).all() else tick_length / 2
             )
             widths.append(np.mean([left_mean, right_mean]) * 2)
-            openness.append(np.isnan(s).sum() / (f).sum())
-            deviations.append(np.nanstd(s))
+
+            f_sum = (f).sum()
+            if not f_sum:
+                openness_score = np.nan
+            else:
+                openness_score = np.isnan(s).sum() / f_sum
+            openness.append(openness_score)
+
+            s_nan = np.isnan(s).all()
+            if s_nan:
+                deviation_score = np.nan
+            else:
+                deviation_score = np.nanstd(s)
+            deviations.append(deviation_score)
```

Comment:
```suggestion
            f_sum = (f).sum()
            s_nan = np.isnan(s)
            if not f_sum:
                openness_score = np.nan
            else:
                openness_score = s_nan.sum() / f_sum
            openness.append(openness_score)

            if s_nan.all():
                deviation_score = np.nan
            else:
                deviation_score = np.nanstd(s)
            deviations.append(deviation_score)
```

This may be better as we call `np.isnan(s)` only once.

Reply:
@martinfleis good call. This was with only one `np.isnan` call, which brings the time down closer to the original. For reference:

* `pysal/main`: 1.71 s ± 31.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
* `jGaboardi:long_scalar_warning`: 1.92 s ± 129 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
  * 2 `np.isnan` calls
* `jGaboardi:long_scalar_warning`: 1.77 s ± 95 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
  * 1 `np.isnan` call